### PR TITLE
client, server: add ScanRegions gRPC protocol support

### PR DESCRIPTION
**Pull request by disksing**
### What problem does this PR solve? 
There are use cases need to iterate all PD regions (for example TiDB resolveLock). It is too slow to query regions one by one.

### What is changed and how it works?
Add `ScanRegions` gRPC method to query regions in batch.

Note: Merge after https://github.com/pingcap/kvproto/pull/396

### Check List 

Tests 
 - Unit test
 - Integration test

Side effects
 - Possible performance regression

Related changes
 - Need to be included in the release notes

### CODE REVIEW

File: client/client_test.go [hunk 1]
```diff
@@ -260,6 +260,53 @@ func (s *testClientSuite) TestGetPrevRegion(c *C) {
...
+		c.Assert(leaders, HasLen, len(expect))
+		for i := range expect {
+			c.Assert(scanRegions[i], DeepEquals, expect[i])
+			c.Assert(leaders[i], DeepEquals, expect[i].Peers[0])
+		}
+	}
+
+	check([]byte{0}, 10, regions)
+	check([]byte{1}, 5, regions[1:6])
+	check([]byte{100}, 1, nil)
```
Comment (MyonKeminta) [inline client/client_test.go]: Can you also test how it works when some regions has no leader?

**Comment by MyonKeminta** [comment]
Rest LGTM

File: client/client_test.go (hunk 1 above)
Comment (disksing) [reply client/client_test.go]: done

File: client/client.go [hunk 2]
```diff
@@ -52,6 +52,9 @@ type Client interface {
 	GetPrevRegion(ctx context.Context, key []byte) (*metapb.Region, *metapb.Peer, error)
 	// GetRegionByID gets a region and its leader Peer from PD by id.
 	GetRegionByID(ctx context.Context, regionID uint64) (*metapb.Region, *metapb.Peer, error)
+	// ScanRegion gets a list of regions, starts from the region that contains key.
+	// Limit limits the maximum number of regions returned.
```
Comment (MyonKeminta) [inline client/client.go]: Can you add comments about the behavior when some regions have no leader?

Comment (disksing) [reply client/client.go]: ok

**Pull request merged**